### DISCUSSION TITLE
Add versioning to orders

### DISCUSSION THREAD

**Discussion by ryanmwc** [craftcms/commerce#2397] [Ideas]
**What are you trying to do?**

My company's business model requires a lot of additional data to be recorded on an order with custom fields, and it's possible these orders will be edited through their lifespan by a 10+ team of people using Commerce. Unfortunately though, there isn't any versioning like with entries to see who last edited them and what they edited. I would imagine that most companies would want to monitor any changes to their orders that are beyond what's added to the order status history.

Since orders are also elements like entries, would it be possible to have a similar feature with a dropdown showing the last 10 or so revisions to the order. Reverting the order to a past state may complicate things, but at least being able to view that state and who made the changes would be helpful.

I see there's a request for versioning with products but didn't see one for orders.

**Comment by bossanova808** [comment]
Audit trail with orders would be great, for sure.

**Comment by lukeholder** [comment]
Definitely want to do this. 

Likely to become a core Craft element feature. Related to https://github.com/craftcms/cms/issues/3946

This would be more like an audit log of changes to an order, than a versioning/undo feature.
